spec-dtslint/operators/tap-spec.ts

```diff
@@ -0,0 +1,20 @@
+import { of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const a = of(1, 2, 3).pipe(tap()); // $ExpectType Observable<number>
+});
+
+it('should accept partial observer', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: number) => x })); // $ExpectType Observable<number>
+  const b = of(1, 2, 3).pipe(tap({ error: (x: number) => x })); // $ExpectType Observable<number>
+  const c = of(1, 2, 3).pipe(tap({ complete: () => 1 })); // $ExpectType Observable<number>
+});
+
+it('should enforce type for next observer function', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: string) => x })); // $ExpectError
+});
+
+it('should accept any type for error observer function', () => {
+  const a = of(1, 2, 3).pipe(tap({ error: (x: string) => x })); // $ExpectType Observable<number>
```

Comment:
Use `error: (err: any) => {}`. The error callback should accept `any` and should not return. TypeScript is okay with it, but the test should reflect the intended usage.